include/bm/bm_sim/target_parser.h

```diff
@@ -108,6 +113,9 @@ class TargetParserBasic final : public TargetParserIface {
   //! Retrieve the value of command-line option `--<name>`. The option needs to
   //! have been previously registered with add_int_option().
   ReturnCode get_int_option(const std::string &name, int *v) const;
+  //! Retrieve the value of command-line option `--<name>`. The option needs to
+  //! have been previously registered with add_uint_option().
+  ReturnCode get_uint_option(const std::string &name, uint32_t *v) const;
```

Comment:
shouldn't this be `unsigned int`? I know you are using `uint32_t` for the port, but using it here doesn't seem very consistent with `[get|add]_int_option` or even with your comment for `add_uint_option`. If you really want to use `uint32_t` here, maybe you should call the methods `add_uint32_option` and `get_uint32_option`.

Reply:
Made it unsigned it.